test/extended/cluster/clt.go

```diff
@@ -0,0 +1,125 @@
+package cluster
+
+import (
+	"fmt"
+	"sync"
+
+	g "github.com/onsi/ginkgo"
+
+	reale2e "k8s.io/kubernetes/test/e2e"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var viperConfig string
+const (
+	DefaultThreadCount int = 10
+	ChannelSize        int = 1024
+)
+
+var _ = g.Describe("[Feature:Performance][Serial][Fast] Load cluster", func() {
+	defer g.GinkgoRecover()
+
+	g.BeforeEach(func() {
+		var err error
+		viperConfig = reale2e.GetViperConfig()
+		e2e.Logf("Using config %v", viperConfig)
+		err = ParseConfig(viperConfig, false)
+		if err != nil {
+			e2e.Failf("Error parsing config: %v", err)
+		}
+	})
+
+	g.It("with templates", func() {
+		project := ConfigContext.ClusterLoader.Projects
+		if project == nil {
+			e2e.Failf("Invalid config file.\nFile: %v", project)
+		}
+
+		numWorkers := ConfigContext.ClusterLoader.Threads
+		if numWorkers == 0 {
+			numWorkers = DefaultThreadCount
+		}
+
+		work := make(chan ProjectMeta, ChannelSize)
+		var wg sync.WaitGroup
+
+		for i := 0; i < numWorkers; i++ {
+			go clWorker(work, &wg)
+		}
+
+		for _, p := range project {
+			for j := 0; j < p.Number; j++ {
+				wg.Add(1)
+				unit := ProjectMeta{j, p, viperConfig}
+				work <- unit
+			}
+		}
+
+		wg.Wait()
+		e2e.Logf("Closing channel")
+		close(work)
+	})
+})
+
+func clWorker(in <-chan ProjectMeta, wg *sync.WaitGroup) {
+	for {
+		p, ok := <-in
+		if !ok {
+			e2e.Logf("Channel closed")
+			return
+		}
+		var oc = exutil.NewCLIWithoutNamespace("cl")
```

Comment:
if you need it threadsafe, you can make it threadsafe.

Reply:
Updated `SetNamespace()` to fix threading issue.